### DISCUSSION TITLE
Access current function from plugin

### DISCUSSION THREAD

**Discussion by deqline** [Vector35/binaryninja-api#1737] [General] [answered]
So I want to be able to access the function which `current_function` gives on binary ninja python console on my plugin. How would I do that?

**Comment by plafosse** [comment, marked as answer]
`current_function` is part of the python console, and thus can't be accessed in a plugin. If you need the current_function you can register a plugin using `register_for_function` which will provide the Function object you want. Either that or you can use the Snippits plugin and create a snippit which does allow access to `current_function`
